pkg/util/flagext/deprecated.go

```diff
@@ -4,10 +4,20 @@ import (
 	"flag"
 
 	"github.com/go-kit/kit/log/level"
+	"github.com/prometheus/client_golang/prometheus"
+	"github.com/prometheus/client_golang/prometheus/promauto"
 
 	"github.com/cortexproject/cortex/pkg/util"
 )
 
+// DeprecatedFlagsUsed is the metric that counts deprecated flags set.
+var DeprecatedFlagsUsed = promauto.NewCounter(
+	prometheus.CounterOpts{
+		Namespace: "cortex",
+		Name:      "deprecated_flags_inuse_total",
```

Comment:
@gouthamve What's the purpose of this metric? I think I'm missing the real use case. When we switch a flag to `flagext.DeprecatedFlag`, that flag is ignored (its value is unused). I understand the policy we want to have for Cortex 1.0 but I think that using `flagext.DeprecatedFlag` for the purpose is not correct, because Cortex will run anyway and deprecated flag ignored.

Reply:
Yep, but we'll have an alert on this and users can then check if they have any deprecated flags set. The user can then check their logs to see which flag has been deprecated.

In the past we've changed flags or features, and or removed flags, by setting the old one to deprecated. And also, this metric is exported, so we can increment in other places where `DeprecatedFlag` is not used.